ouroboros-consensus/src/Ouroboros/Consensus/Block/Forging.hs

```diff
@@ -173,16 +176,38 @@ data MaxTxCapacityOverride
     -- are invalid according to the ledger and we want certainly don't want to
     -- forge invalid blocks.
 
+type family Overrides blk
+
 computeMaxTxCapacity ::
-     LedgerSupportsMempool blk
+     forall blk . (TxLimits blk, Overrides blk ~ Measure blk)
   => TickedLedgerState blk
-  -> MaxTxCapacityOverride
-  -> Word32
+  -> MaxTxCapacityOverride blk
+  -> Measure blk
 computeMaxTxCapacity ledger maxTxCapacityOverride = case maxTxCapacityOverride of
       NoMaxTxCapacityOverride     -> noOverride
-      MaxTxCapacityOverride txCap -> noOverride `min` txCap
+      MaxTxCapacityOverride txCap -> noOverride `minMeasure` txCap
   where
-    noOverride = maxTxCapacity ledger
+    noOverride = maxCapacity ledger
+    minMeasure x y
+      | lessEq @blk x y = x
```

Comment:
This logic still seems too fragile to me. If the user accidentally sets one override above the ledger state's limit, then all of their overrides will be ignored.

Reply:
Thinking to address it separately, issue defined here https://github.com/input-output-hk/ouroboros-network/issues/3239